pkg/webhook/certs_test.go

```diff
@@ -48,48 +58,63 @@ func TestCertExpiry(t *testing.T) {
 }
 
 func TestBadCA(t *testing.T) {
-	caArtifacts, err := createCACert()
+	cr := &certRotator{
+		caName:         caName,
+		caOrganization: caOrganization,
+		dnsName:        dnsName,
+	}
+	caArtifacts, err := cr.createCACert()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cert, key, err := createCertPEM(caArtifacts)
+	cert, key, err := cr.createCertPEM(caArtifacts)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	badCAArtifacts, err := createCACert()
+	badCAArtifacts, err := cr.createCACert()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if validServerCert(badCAArtifacts.CertPEM, cert, key) {
+	if cr.validServerCert(badCAArtifacts.CertPEM, cert, key) {
 		t.Error("Generated cert is valid when it should not be")
 	}
 }
 
 func TestSelfSignedCA(t *testing.T) {
-	caArtifacts, err := createCACert()
+	cr := &certRotator{
+		caName:         caName,
+		caOrganization: caOrganization,
+		dnsName:        dnsName,
+	}
+	caArtifacts, err := cr.createCACert()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if !validCACert(caArtifacts.CertPEM, caArtifacts.KeyPEM) {
+	if !cr.validCACert(caArtifacts.CertPEM, caArtifacts.KeyPEM) {
 		t.Error("Generated cert is not valid")
 	}
 }
 
 func TestCAExpiry(t *testing.T) {
-	caArtifacts, err := createCACert()
+	cr := &certRotator{
+		caName:         caName,
+		caOrganization: caOrganization,
+		dnsName:        dnsName,
+	}
+	caArtifacts, err := cr.createCACert()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if !validCACert(caArtifacts.CertPEM, caArtifacts.KeyPEM) {
+	if !cr.validCACert(caArtifacts.CertPEM, caArtifacts.KeyPEM) {
 		t.Error("Generated cert is not valid")
 	}
 
-	valid, err := validCert(caArtifacts.CertPEM, caArtifacts.CertPEM, caArtifacts.KeyPEM, DNSName, time.Now().Add(11*365*24*time.Hour))
```

Comment:
I think this was a small bug here to use `DNSName` constant although the test result is same. The `err != nil` should because the name doesn't match.

I changed it to `caName` PTAL.

Reply:
Thanks for catching this!